wandb/sdk/artifacts/artifact.py

```diff
@@ -862,6 +876,12 @@ def _populate_after_save(self, artifact_id: str) -> None:
                 "ttlIsInherited",
                 "",
             )
+
+        if "tags" in fields:
+            query_template = query_template.replace("_MAYBE_TAGS_", "tags {name}")
+        else:
+            query_template = query_template.replace("_MAYBE_TAGS_", "")
```

Comment:
i see it's done this way in this function already, but i have to wonder: why not use a format string?

```python
if "ttlIsInherited" in fields:
    query_ttl_duration_seconds = "ttlDurationSeconds"
    query_ttl_inherited = "ttlIsInherited"
else:
    query_ttl_duration_seconds = ""
    query_ttl_inherited = ""

query_tags = "tags {name}" if "tags" in fields else ""
    
query_template = f"""
    query ArtifactByIDShort($id: ID!) {{
        artifact(id: $id) {{
            versionIndex
            {query_ttl_duration_seconds}
            {query_ttl_inherited}
            ...
            {query_tags}
            state
            ...
        }}
    }}
"""
```

Reply:
> i see it's done this way in this function already  

This is the main reason TBH 🙂 

Secondary reason is a little more subjective/cosmetic -- doubled curly braces `{{`/`}}` would be required over multiple levels, as you mention ☝️ .  I feel like if we were to rewrite this part (in a separate PR), we might want to consider building the query programmatically instead -- e.g. via [gql.dsl](https://gql.readthedocs.io/en/stable/advanced/dsl_module.html) or maybe another alternative

Reply:
Makes sense! Yeah using the DSL would be even better. I'm just surprised by this string-replace stuff, it's very non-Pythonic.